source/MaterialXView/Main.cpp

```diff
@@ -245,12 +272,17 @@ int main(int argc, char* const argv[])
         viewer->setEnvSampleCount(envSampleCount);
         viewer->setLightRotation(lightRotation);
         viewer->setDocumentModifiers(modifiers);
+        viewer->initialize();
         if (!captureFilename.empty())
         {
             viewer->requestFrameCapture(captureFilename);
             viewer->requestExit();
         }
-        viewer->initialize();
+        else if(!bakeFilename.empty()) 
+        {
+            viewer->bakeTextures();
+            viewer->requestExit();
+        }
         ng::mainloop(refresh);
     }
     ng::shutdown();
```

Comment:
I'm seeing some unintended modifications to NanoGUI below, and make sure to revert these!

Reply:
NanoGUI on master cannot be fully fetched because one of the objects in the git for pybind was not updated.